ext/src/ruby_api/store.rs

```diff
@@ -300,6 +300,12 @@ impl<'a> StoreContextValue<'a> {
         } else if let Some(exit) = error.downcast_ref::<I32Exit>() {
             wasi_exit_error().new_instance((exit.0,)).unwrap().into()
         } else {
+            if let Some(error) = error.downcast_ref::<std::io::Error>() {
+                if error.kind() == std::io::ErrorKind::WriteZero {
+                    return error!("{}", error.to_string());
+                }
+            }
+
```

Comment:
I think we can leave this out and let the error flow to `Trap::from_error`, as it will internally take care of the generic case and any Wasm traps. According to the docs, though, any error in the host, will trigger a Wasm trap https://docs.wasmtime.dev/api/wasmtime/struct.Func.html#errors.

Reply:
sure, if it is fine to just raise a wasm trap in this case then I can let it do that